### PR TITLE
Add support for Python 3

### PR DESCRIPTION
**Pull request by kylef**
This pull request adds support for Python 3 to this library.

Would appreciate if you can review and make a release with Python 3 support @richard-jones.

### CODE REVIEW

**Comment by richard-jones** [comment]
Hi @kylef 

Thanks for the PR - we'll review and merge as soon as we can.

Glad you are getting some use out of the library!

File: negotiator/negotiator.py [hunk 1]
```diff
@@ -407,7 +407,7 @@ def negotiate(self, accept=None, accept_language=None, accept_encoding=None, acc
         - accept_charset - HTTP Header: Accept-Charset; not currently supported in negotiation
         - accept_packaging - HTTP Header: Accept-Packaging (from SWORD 2.0); a URI only, no q values
         
-        If verbose=True, then this will print to stdout
+        If verbose=True, then this will print(to stdout
```
Comment (honzajavorek) [inline negotiator/negotiator.py]: This looks like mistake in search & replace :)

Comment (kylef) [reply negotiator/negotiator.py]: Oh yeah, I've fixed this and updated PR. Thank you for reviewing it.

**Comment by kylef** [comment]
@richard-jones Would I be able to get you to cut a release adding these changes please. Would really appreciate it (let me know if I can help in any way).

**Comment by richard-jones** [comment]
I had a go after I merged the PR, but something seems to have changed on pypi and the instructions I usually follow don't seem to work - something about a missing legacy URL, which is odd, since the pypi instructions still mention it.  I haven't had time to debug it, and if you have an up-to-date set of release instructions to hand that would be useful!

**Comment by kylef** [comment]
I think this may be due to `url` property in setup.py being set to an empty string, perhaps this can be set to `https://github.com/CottageLabs/negotiator`.

Regarding instructions, I'd say the current best practises are documented at https://packaging.python.org/tutorials/distributing-packages/#packaging-your-project and it's recommended/encouraged to upload wheel along with using twine. `python setup.py upload` is strongly discouraged due as it may use a plaintext HTTP or unverified HTTPS connection on some Python versions, allowing your username and password to be intercepted during transmission.

So:

```shell
# Install twine and wheel
$ pip install wheel twine

# Create the source distribution and universal wheel
$ python setup.py sdist  bdist_wheel --universal

# Upload distribution using twine
$ twine upload dist/*
```

**Comment by richard-jones** [comment]
Yeah, that's roughly what I tried.  I get the error:

SSLError: HTTPSConnectionPool(host='upload.pypi.org', port=443): Max retries exceeded with url: /legacy/ (Caused by SSLError(SSLError(1, '_ssl.c:510: error:14077438:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert internal error'),))

I have a .pypirc file which just contains my un/pw, so I'm not sure where it's getting the /legacy url.  I have tried adding a "repository" line with a variety of pypi upload urls from documentation, and always the same error about the /legacy url

**Comment by kylef** [comment]
According to that error it seems that SSLv2/3 is used which it seems that the upload API does not support (for good reasons, it has weak security and TLSv1.2 is often preferred).

```
$ curl https://upload.pypi.org -3 -I
curl: (35) Unknown SSL protocol error in connection to upload.pypi.org:-9836
```

```
$ curl https://upload.pypi.org -1 -I
HTTP/1.1 200 OK
Connection: keep-alive
Server: TwistedWeb/17.9.0
Date: Mon, 16 Oct 2017 10:13:16 GMT
Content-Type: text/html; charset=UTF-8
Content-Length: 2637
Etag: "Oe6SC2xhk7oqnd3tI5GBNg"
Content-Security-Policy: base-uri 'self'; block-all-mixed-content; connect-src 'self' https://2p66nmmycsj3.statuspage.io; default-src 'none'; font-src 'self' fonts.gstatic.com; form-action 'self'; frame-ancestors 'none'; frame-src 'none'; img-src 'self' https://warehouse-camo.herokuapp.com/ https://secure.gravatar.com; script-src 'self' www.google-analytics.com; style-src 'self' fonts.googleapis.com
Referrer-Policy: origin-when-cross-origin
Vary: Accept-Encoding
Via: 1.1 vegur
```

I am not sure about your setup/environment, but perhaps upgrading Python and/or OpenSSL can allow it to use TLS 1.2 or TLS 1.3 which is supported.

**Comment by kylef** [comment]
If you're having difficulties, I think you can log into the PyPI website and upload the sdist/bdist wheel manually too: https://pypi.python.org/pypi

**Pull request merged**